### PR TITLE
Allura hvl bug

### PR DESCRIPTION
**Pull request by MaxHellstrom**
solves #31. 

Started as a suspected bug fix in the dose calculation for Philips SR data. It turned out to be a problem with incorrect parsing of multiple X-ray filter materials.

In addition to the above, some missing imports have been added (black, isort) as well as other small improvements in code readability.

### CODE REVIEW

**Comment by BwKodex** [comment]
It would help if your merge request description was a bit more descriptive of what this change does :)

File: src/pyskindose/rdsr_normalizer.py [hunk 1]
```diff
@@ -5,6 +5,15 @@
 import numpy as np
 import pandas as pd
 
+from .constants import (
```
Comment (BwKodex) [inline src/pyskindose/rdsr_normalizer.py:8]: Good use of constants!!!

File: src/pyskindose/rdsr_normalizer.py [hunk 2]
```diff
@@ -217,10 +226,47 @@ def _normalize_beam_parameters(
     data_norm["kVp"] = data_parsed.KVP_kV
     data_norm["K_IRP"] = data_parsed.DoseRP_Gy * 1000
 
-    data_norm["filter_thickness_Cu"] = data_parsed.XRayFilterThicknessMaximum_mm
+    # Ad filter thincknesses...
```
Comment (BwKodex) [inline src/pyskindose/rdsr_normalizer.py:229]: Why ...?

File: src/pyskindose/rdsr_normalizer.py [hunk 3]
```diff
@@ -217,10 +226,47 @@ def _normalize_beam_parameters(
...
-    data_norm["filter_thickness_Al"] = [0.0] * len(data_parsed.XRayFilterThicknessMaximum_mm)
+    # Add columns for filter materials, and initialize to zero
+    for key in [
+        KEY_NORMALIZATION_FILTER_SIZE_COPPER,
+        KEY_NORMALIZATION_FILTER_SIZE_ALUMINUM,
+    ]:
+        data_norm[key] = 0.0
+
+    # for each irradiation event
+    for event_index in range(len(data_parsed)):
```
Comment (BwKodex) [inline src/pyskindose/rdsr_normalizer.py:243]: I feel like this/the contents of this for loop could be broken out into a separate function for more clarity when reading the code.

**Review by BwKodex — CHANGES_REQUESTED** [review]
Test/-s should be added for validating the new parsing functionality.

File: src/pyskindose/normalization_settings.json [hunk 4]
```diff
@@ -54,9 +52,12 @@
...
                 "At3": "+"
             },
-
             "field_size_mode": "ASD",
-            "detector_side_length": 40
+            "detector_side_length": 40,
+            "filter_materials": [
+                "Cu",
+                "Al"
+            ]
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply src/pyskindose/normalization_settings.json]: yes.  Forgot to remove it. Fixed now

File: src/pyskindose/normalization_settings.json [hunk 5]
```diff
@@ -25,9 +23,11 @@
                 "At2": "+",
                 "At3": "+"
             },
-
             "field_size_mode": "CFA",
-            "detector_side_length": 40
+            "detector_side_length": 40,
+            "filter_materials": [
+                "Cu"
+            ]
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply src/pyskindose/normalization_settings.json]: fixed

File: tests/unittests/test_rdsr_parsing.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    data_norm_siemens = rdsr_normalizer(data_parsed=data_parsed_siemens)
+
+    test.append(data_norm_philips["filter_thickness_Al"][0])
+    test.append(data_norm_philips["filter_thickness_Cu"][0])
+    test.append(data_norm_siemens["filter_thickness_Al"][0])
+    test.append(data_norm_siemens["filter_thickness_Cu"][0])
+
+    a = 1
+
+    assert expected == test
```
Comment (BwKodex) [inline tests/unittests/test_rdsr_parsing.py:50]: you should switch order on these 2. Pytest assumes that the actual value is the first

File: tests/unittests/test_rdsr_parsing.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    # parse raw rdsr file
+    data_parsed = rdsr_parser(data_raw=RDSR_file)
+
+    test = len(data_parsed)
+
+    assert expected == test
+
+
+def test_xray_filter_material_parsing():
```
Comment (BwKodex) [inline tests/unittests/test_rdsr_parsing.py:29]: Kind of nitpicky but you could be a bit more explicit in the name with what you're testing. What about the parsing is tested here? That any data is extracted. That both an aluminium and copper filter is extracted?

**Review by BwKodex — CHANGES_REQUESTED** [review]
Great with more tests! :)

File: src/pyskindose/rdsr_normalizer.py (hunk 3 above)
Comment (MaxHellstrom) [reply src/pyskindose/rdsr_normalizer.py:233]: done

File: tests/unittests/test_rdsr_parsing.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+from pyskindose.rdsr_normalizer import rdsr_normalizer
+from pyskindose.rdsr_parser import rdsr_parser
+
+phantom_path = Path(__file__).parent.parent.parent / "src" / "pyskindose" / "example_data" / "RDSR"
+
+
+def test_nr_irradiation_events_parsing_in_rdsr_parser():
+
+    # load SR file with known number of irradiation events
+    RDSR_file = pydicom.read_file(phantom_path / "s1.dcm")
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply tests/unittests/test_rdsr_parsing.py]: yes! Fixed.

File: tests/unittests/test_rdsr_parsing.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+    test = len(data_parsed)
+
+    assert expected == test
+
+
+def test_xray_filter_material_parsing():
+    # knowns filter thicknesses
+    expected = [1.0, 0.4, 0.0, 0.6]
+    test = []
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply tests/unittests/test_rdsr_parsing.py:32]: done

File: src/pyskindose/table_data/HVL_simulated.csv [hunk 10]
```diff
@@ -455,915 +455,915 @@ kVp_kV,InherentFiltration_mmAl,AddedFiltration_mmCu,AddedFiltration_mmAl,AddedFi
 125,4.1,0.3,0,770,AXIOM-Artis,Single Plane,8,9.33,0.885,Inherent filtration measured in lab 106 - 2018-03-28
 125,4.1,0.6,0,770,AXIOM-Artis,Single Plane,8,10.7,0.929,Inherent filtration measured in lab 106 - 2018-03-28
 125,4.1,0.9,0,770,AXIOM-Artis,Single Plane,8,11.5,0.948,Inherent filtration measured in lab 106 - 2018-03-28
-50,4.4,0,0,795,AlluraClarity,Plane A,11,2.37,0.812,Inherent filtration measured in lab U104 - 2018-04-12
```
[the remark this answers is not in the document]
Comment (MaxHellstrom) [reply src/pyskindose/table_data/HVL_simulated.csv:458]: great idea. I have added an issue for this, but I wont address it in this PR.

**Review by BwKodex — APPROVED** [review]

**Pull request merged**